vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/SampleOverviewComponent.java

```diff
@@ -352,6 +372,48 @@ private void setupSearchFieldForExperimentTabs(String experimentName,
         }
       });
     }
+
+    private Result<BatchId, BatchRegistrationService.ResponseCode> registerBatch(
+        BatchRegistrationContent batchRegistrationContent) {
+      return batchRegistrationService.registerBatch(batchRegistrationContent.batchLabel(),
+          batchRegistrationContent.isPilot());
+    }
+
+    private Result<List<life.qbic.projectmanagement.domain.project.sample.Sample>, SampleRegistrationService.ResponseCode> registerSamplesForBatch(
+        BatchId batchId, List<SampleRegistrationContent> sampleRegistrationContents) {
+      List<SampleRegistrationRequest> sampleRegistrationRequests = createSampleRegistrationRequests(
+          batchId, sampleRegistrationContents);
+      return sampleRegistrationService.registerSamples(sampleRegistrationRequests, projectId);
+    }
+
+    private List<SampleRegistrationRequest> createSampleRegistrationRequests(BatchId batchId,
+        List<SampleRegistrationContent> sampleRegistrationContent) {
+      List<SampleRegistrationRequest> sampleRegistrationRequests = new ArrayList<>();
+      for (SampleRegistrationContent registrationContent : sampleRegistrationContent) {
+        Analyte analyte = new Analyte(registrationContent.analyte());
+        Specimen specimen = new Specimen(registrationContent.specimen());
+        Species species = new Species(registrationContent.species());
+        BiologicalReplicateId biologicalReplicateId = BiologicalReplicateId.create();
+        SampleOrigin sampleOrigin = SampleOrigin.create(species, specimen, analyte);
+        SampleRegistrationRequest sampleRegistrationRequest = new SampleRegistrationRequest(
+            registrationContent.label(), batchId, experimentId,
+            registrationContent.experimentalGroupId(), biologicalReplicateId, sampleOrigin);
+        sampleRegistrationRequests.add(sampleRegistrationRequest);
```

Comment:
i see quite some potential for code readability here.

For example you can iterate via the Java stream API over the registration content and call a method that then actually creates a single registration request (passing the batchId additionally everytime, but no problem, you have anyway).

You will see the resulting code is much more readable.

(Always ask: what does this function do? If it does more than one thing, you probably should extract some code into an own function, and then make a composition of functions).

Here the top function's task is: Iterate through the list of registration content and create a request for each, given a common batch Id.

The child function that is called in the loop will do then one request per element.

Reply:
Good point 👍